api/routes/books.js

```diff
@@ -28,9 +33,22 @@ router.get('/:id', async (req, res, next) => {
 router.post('/', async (req, res, next) => {
   const status = 200
   try {
+    // Authorization first
+    const token = req.headers.authorization.split('Bearer ')[1]
+    const payload = jsonwebtoken.verify(token, SECRET_KEY)
+
+    // Authentication
+    const user = await User.findOne({ _id: payload.id })
+
+    const isAdmin = user.admin
+
+    // Error if not admin
+    if (!isAdmin) throw new Error(`Access Denied: Must be Admin`)
```

Comment:
I think some of these authentication and authorization checks could be done via middleware!